src/axios-services/index.js

```diff
@@ -120,20 +236,237 @@ export const createPendingOrder = async (email, address) => {
 export const fetchAllProducts = async () => {
   try {
     const { data: products } = await axios.get(`${BASE_URL}/products`);
-    console.log('all products: ', products);
+
     return products;
-  } catch(error) {
+  } catch (error) {
     throw error;
   }
-}
+};
+
+export const fetchAllOrders = async () => {
+  try {
+    const { data: orders } = await axios.get(`${BASE_URL}/orders`);
+
+    return orders;
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const fetchSingleProduct = async (id) => {
+  try {
+    const { data: product } = await axios.get(
+      `${BASE_URL}/products/productid/${id}`
+    );
+
+    return product;
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const updateProduct = async (
+  token,
+  { id, name, photo, description, price, category, quantity }
+) => {
+  try {
+    const { data: product } = await axios.patch(
+      `${BASE_URL}/products/${id}`,
+      { name, photo, description, price, category, quantity },
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    );
+
+    return product;
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const fetchCategory = async (category) => {
+  try {
+    const { data: products } = await axios.get(
+      `${BASE_URL}/products/categories/${category}`
+    );
+
+    return products;
+  } catch (error) {
+    throw error;
+  }
+};
 
 export const fetchUserOrder = async (username) => {
   try {
     const { data: userOrder } = await axios.get(
-      `${BASE_URL}/orders/username/:username`
+      `${BASE_URL}/orders/username/${username}`
     );
     return userOrder;
   } catch (error) {
     console.error(error);
   }
-};
\ No newline at end of file
+};
+
+export const addProductToCart = async (orderId, productId) => {
+  try {
+    const { data: cartProduct } = await axios.post(
+      `
+      ${BASE_URL}/products_orders`,
+      {
+        orderId,
+        productId,
+        quantity: 1,
+      }
+    );
+    return cartProduct;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const fetchProductOrderById = async (orderId) => {
```

Comment:
just be careful of the naming because you're grabbing by the order id, i would rename to fetch productorderbyorderid